mne/decoding/ems.py

```diff
@@ -76,38 +179,36 @@ def compute_ems(epochs, conditions=None, picks=None, n_jobs=1, verbose=None):
                          len(conditions))
 
     ev = epochs.events[:, 2]
-    # special care to avoid path dependent mappings and orders
+    # Special care to avoid path dependent mappings and orders
     conditions = list(sorted(conditions))
     cond_idx = [np.where(ev == epochs.event_id[k])[0] for k in conditions]
 
     info = pick_info(epochs.info, picks)
     data = epochs.get_data()[:, picks]
 
     # Scale (z-score) the data by channel type
+    # XXX the z-scoring is applied outside the CV! Not standard!
     for ch_type in ['mag', 'grad', 'eeg']:
         if ch_type in epochs:
+            # FIXME should be applied to all sort of data channels
             if ch_type == 'eeg':
                 this_picks = pick_types(info, meg=False, eeg=True)
             else:
                 this_picks = pick_types(info, meg=ch_type, eeg=False)
             data[:, this_picks] /= np.std(data[:, this_picks])
 
-    try:
-        from sklearn.model_selection import LeaveOneOut
-    except:  # XXX support sklearn < 0.18
-        from sklearn.cross_validation import LeaveOneOut
-
-    def _iter_cv(n):  # XXX support sklearn < 0.18
-        if hasattr(LeaveOneOut, 'split'):
-            cv = LeaveOneOut()
-            return cv.split(np.zeros((n, 1)))
-        else:
-            cv = LeaveOneOut(len(data))
-            return cv
+    # Setup cross-validation. Need to use _set_cv to deal with sklearn
+    # deprecation of cv objects.
+    y = epochs.events[:, 2]
+    _, cv_splits = _set_cv(cv, clf=LogisticRegression(), X=y, y=y)
 
     parallel, p_func, _ = parallel_func(_run_ems, n_jobs=n_jobs)
+    # FIXME this parallization should be removed.
+    #   1) it's numpy computation so it's already efficient,
+    #   2) it duplicates the data in RAM,
+    #   3) the computation is already super fast.
```

Comment:
Agree. This is all trivial linalg stuff that takes no time. My hunch would be the parallelisation slows it down for most cases.